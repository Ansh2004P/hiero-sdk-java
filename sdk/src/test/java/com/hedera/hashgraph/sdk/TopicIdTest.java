package com.hedera.hashgraph.sdk;

import io.github.jsonSnapshot.SnapshotMatcher;

import com.google.protobuf.InvalidProtocolBufferException;

import org.junit.AfterClass;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

class TopicIdTest {
    @BeforeAll
    public static void beforeAll() {
        SnapshotMatcher.start();
    }

    @AfterClass
    public static void afterAll() {
        SnapshotMatcher.validateSnapshots();
    }

    @Test
    void shouldSerializeFromString() {
        SnapshotMatcher.expect(TopicId.fromString("0.0.5005").toProtobuf().toString()).toMatchSnapshot();
    }

    @Test
    void toBytes() throws InvalidProtocolBufferException {
        SnapshotMatcher.expect(new TopicId(5005).toBytes()).toMatchSnapshot();
    }

    @Test
    void fromBytes() throws InvalidProtocolBufferException {
        SnapshotMatcher.expect(TopicId.fromBytes(new TopicId(5005).toBytes())).toMatchSnapshot();
    }
}
