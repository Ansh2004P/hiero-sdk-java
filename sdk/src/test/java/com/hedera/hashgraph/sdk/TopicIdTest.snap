com.hedera.hashgraph.sdk.TopicIdTest.fromBytes=[
  {
    "shard": 0,
    "realm": 0,
    "num": 5005
  }
]


com.hedera.hashgraph.sdk.TopicIdTest.shouldSerialize=[
  "# com.hedera.hashgraph.sdk.proto.TopicID@bc8a0\nrealm_num: 0\nshard_num: 0\ntopic_num: 5005"
]


com.hedera.hashgraph.sdk.TopicIdTest.shouldSerializeFromString=[
  "# com.hedera.hashgraph.sdk.proto.TopicID@bc8a0\nrealm_num: 0\nshard_num: 0\ntopic_num: 5005"
]


com.hedera.hashgraph.sdk.TopicIdTest.toBytes=[
  "GI0n"
]